test/include/seqan3_test/tmp_filename.hpp

```diff
@@ -0,0 +1,160 @@
+ * It automatically removes the temporary directory and all contained files and subdirectories on destruction.
+ *
+ * \par Example
+ *
+ * ```cpp
+ * tmp_file_name fn{"my_file"};
+ * std::cout << fn.get_path() << std::endl;
+ * ```
+ *
+ * \par Exceptions
+ *
+ * Might throw a std::filesystem::filesystem_error on failure to create a temporary file directory.
+ *
+ * \par Thread safety
+ *
+ *  According to "https://www.gnu.org/software/libc/manual/html_node/Temporary-Files.html" the call to
+ * \a mkdtemp is thread safe, such that creating multiple parallel instances of this class will
+ * not induce a data race on the creation of temporary file path.
+ */
+class tmp_file_name
+{
+public:
+
+    /* rule of six */
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    //!\brief Deleted default constructor.
+    tmp_file_name() = delete;
+    //!\brief Copy constructor.
+    tmp_file_name(tmp_file_name const &) = default;
+    //!\brief Move constructor.
+    tmp_file_name(tmp_file_name &&) = default;
+    //!\brief Copy assignment.
+    tmp_file_name & operator=(tmp_file_name const &) = default;
+    //!\brief Move assignment.
+    tmp_file_name & operator=(tmp_file_name &&) = default;
+
+    /*!\brief Constructs temp path with given file name.
+     * \param f_name The name of the file.
+     *
+     * The generated file name is unique due to a call to \a mkdtemp.
+     *
+     * \par Exceptions
+     * Might throw std::filesystem::filesystem_error.
+     */
+    explicit tmp_file_name(const char * f_name)
+    {
+        if (f_name == nullptr)
+            throw fs::filesystem_error("Empty file name!",
+                                        std::make_error_code(std::errc::invalid_argument));
+
+        auto tmp_base_dir = fs::temp_directory_path();
+        tmp_base_dir /= fs::path{"seqan_test_XXXXXXXX"};
+        // We have to use mkdtemp, which is not deprecated. We place it into the dedicated tmp_dir
+        // returned by temp_directory_path. Within this path we can safely create files, that would be
+        // unique per test instance as the parent directory is.
+        auto path_str = tmp_base_dir.string();  // Copy the underlying path to get access to the raw char *.
+        if (char * f = mkdtemp(path_str.data()); f != nullptr)  // mkdtemp replaces XXXXXXXX in a safe and unique way.
+        {
+            file_path = f;
+            file_path /= fs::path{f_name};
+            return;
+        }
+        throw fs::filesystem_error("Could not create temporary directory with mkdtemp!",
+                                   tmp_base_dir,
+                                   std::make_error_code(std::errc::bad_file_descriptor));
+    }
+
+    /*!\brief Destructs the temporary file path.
+     * Removes the temporary directory and all it's subdirectories and files contained.
+     */
+    ~tmp_file_name()
+    {
+        [[maybe_unused]] std::error_code ec;
+        remove_all(file_path.parent_path(), ec);
```

Comment:
Is `remove_all` in the seqan namespace?!

Reply:
No, it is in std namespace. So usually I would suggest to put an `using std::remove_all` inside of it, but in this case, I don't think what we will use something generic for the`path` object. So the lookup system should find it, right?